habitat-lab/habitat/tasks/rearrange/social_nav/social_nav_sensors.py

```diff
@@ -0,0 +1,534 @@
+            ]
+            / self._val_dict["min_start_end_episode_step"],
+            "follow_human_steps_after_frist_encounter": self._val_dict[
+                "after_found_human_times"
+            ],
+            "follow_human_steps_ratio_after_frist_encounter": self._val_dict[
+                "after_found_human_times"
+            ]
+            / (
+                self._total_step - self._val_dict["min_start_end_episode_step"]
+            ),
+            "backup_ratio": self._val_dict["backup_count"]
+            / self._val_dict["step"],
+            "yield_ratio": self._val_dict["yield_count"]
+            / self._val_dict["step"],
+        }
+
+        # Update the counter
+        if self._val_dict["has_found_human"]:
+            self._val_dict["step_after_found"] += 1
+
+
+@registry.register_measure
+class SocialNavSeekSuccess(Measure):
+    """Social nav seek success meassurement"""
+
+    cls_uuid: str = "nav_seek_success"
+
+    @staticmethod
+    def _get_uuid(*args, **kwargs):
+        return SocialNavSeekSuccess.cls_uuid
+
+    def reset_metric(self, *args, task, **kwargs):
+        """Reset the metrics"""
+        task.measurements.check_measure_dependencies(
+            self.uuid,
+            [NavToPosSucc.cls_uuid, RotDistToGoal.cls_uuid],
+        )
+        self._following_step = 0
+        self.update_metric(*args, task=task, **kwargs)
+
+    def __init__(self, *args, config, sim, **kwargs):
+        self._config = config
+        self._sim = sim
+
+        super().__init__(*args, config=config, **kwargs)
+        # Setup the parameters
+        self._following_step = 0
+        self._following_step_succ_threshold = (
+            config.following_step_succ_threshold
+        )
+        self._safe_dis_min = config.safe_dis_min
+        self._safe_dis_max = config.safe_dis_max
+        self._use_geo_distance = config.use_geo_distance
+        self._need_to_face_human = config.need_to_face_human
+        self._facing_threshold = config.facing_threshold
+        self._robot_idx = config.robot_idx
+        self._human_idx = config.human_idx
+
+    def update_metric(self, *args, episode, task, observations, **kwargs):
+        # Get the angle distance
+        angle_dist = task.measurements.measures[
+            RotDistToGoal.cls_uuid
+        ].get_metric()
+
+        # Get the positions of the human and the robot
+        use_k_human = f"agent_{self._human_idx}_localization_sensor"
+        human_pos = observations[use_k_human][:3]
+        use_k_robot = f"agent_{self._robot_idx}_localization_sensor"
+        robot_pos = observations[use_k_robot][:3]
+
+        # If we want to use the geo distance
+        if self._use_geo_distance:
+            dist = self._sim.geodesic_distance(robot_pos, human_pos)
+        else:
+            dist = task.measurements.measures[DistToGoal.cls_uuid].get_metric()
+
+        # Compute facing to human
+        vector_human_robot = human_pos - robot_pos
```

Comment:
let's make facing human a function. This code is repeated many times in these functions.

Reply:
Thanks, you are correct! I corrected this!